### DISCUSSION TITLE
Exputt speed?

### DISCUSSION THREAD

**Discussion by jmjava** [springbok/MLM2PRO-GSPro-Connector#67] [General]
I was able to get Exputt registering putts in practice mode but it really seemed to be off as far as distance goes.  I did a quick look at the API log and it appeared the speeds were coming in as on the screen.  Are the units in Exputt the same as expected in GSPro?  I had to hit putts like 3x as hard.  It may be my setup I had 0.27 and had to revert back to 0.20 to get the verify working for Exputt

**Comment by springbok** [comment]
What have you got the units set to?

Can you post your ROI's?

Can you send me the log file.

**Reply by jmjava** [reply]
@springbok thanks for the quick reply.  I think the "units" was the issue.  I had to hit it much much harder when it was set to meters now its set to feet and it appears to be more reasonable.  I also re-synced with .27 so that could also be part of it... anyway it appears better now

![image](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/591751/d953a562-e9f3-43b6-ba45-ad4d5de75b9e)

**Comment by wonder99** [comment]
Hi,
so Exputt and GSPro use different calculations to turn ball speed into a
resulting distance.  I discovered this, and analyzed it pretty deeply.  I
never got an answer why they are different, but the GSPRO guys said they
actually measured putts to create those tables.
My problems with their calculations are:
1. The distance an object will roll is a function of the initial speed and
the friction of the ground (which is derived from the stimp).  At a 4.1mph
launch speed (10 feet per second), the distance travelled should be the
number of feet equal to the stimp (10 ft on a stimp 10 green).  That's the
definition of stimp and it's true for exputt but not for gspro.  They are
close.   As the speeds go up, they diverge quite a bit.
2. Regardless of the stimp, a putt which starts at twice the speed should
roll 4 times the distance.  That's just physics.  I don't know why both
tables don't follow that relationship, but they don't.

Note: I ignore a tiny loss which is caused by the need to get the ball
rotating (angular momentum).  This will cause your putt to be a little
shorter than the simple physics would predict.  However, this is offset by
the fact that for the first little part of the putt, there is far less
friction because the ball isn't in complete contact with the ground. And
the faster putt, the more energy needs to be converted to spin, but also
there's more 'flight' time.  So I'd say they basically cancel out.

Here's the analysis I did
https://docs.google.com/spreadsheets/d/1AifoFPqQGdLLMzLQYZZskcix0DE6yIlzt1sySM8QQac/edit#gid=0

Bottom line is yes, the two programs will yield different results, even
under the same conditions.  Within about 12-15 feet, they are basically
close enough.  But the farther you go, the wilder it gets.  Also notice
that Exputt is quite generous with distance on slow putts.
- Paul

**Reply by jmjava** [reply]
@wonder99 wow that's pretty impressive analysis :)  I guess I am in the right place to learn here :)  As long as its its similar within 0-15 that should be good enough for me.   

This is awesome!  I guess this is saying Exputt has a bit less error vs the calculated distance?

![image](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/591751/e893f367-2f32-4ef8-a7bb-96b07ec49c83)

**Comment by wonder99** [comment]
I went overboard with that response, because now I see you're seeing 3x
difference.  Has to be a feet / yards or meters units problem as Etienne
said.  I think you need to set exputt in mph.
- Paul

**Reply by trplbgy** [reply]
Should also consider these tables and distance traveled based on launch speed alone is inexact as there are factors that have to be "averaged" or taken out of consideration such as grass type, green slope, firmness of the surface, and golfer technique, all of which can affect the ball's roll. This doesn't help the answer as I've searched for this as well but is why software have different reasonings for the outputs. Sometimes they change it for user experience also.... as it can be hard to "feel" indoors..

**Reply by wonder99** [reply]
Well, for all of this we know the slope is perfectly flat--that's what the tables are reporting, as well as the calculations.  Firmness and grass type are rolled into the resulting stimp, so those are also accounted for.  Once the ball has started rolling at a speed, the golfer's technique is no longer a factor, except the tiny launch angle and tiny initial spin.  Exputt doesn't do anything to measure those anyway, it only knows launch direction and initial speed.  The distance that results comes from just those two things.
In any case, none of this matters much since it's really hard to create accurate tables (who has a perfectly flat and consistent putting surface that's 60+ ft?)  And in the end, it's just a game so we just need do the best with what it has assumed.  
As you said, in the real world and even in sim, no green is perfectly flat for that distance, so you're always dealing with differences.  I just felt that the tables should be so ideal, much of that uncertainty would be removed.
